ztp/policygenerator/policyGen/policyBuilder.go

```diff
@@ -333,7 +333,12 @@ func (pbuilder *PolicyBuilder) splitYamls(yamls []byte) ([][]byte, error) {
 		if err != nil {
 			return nil, err
 		}
-		resources = append(resources, resBytes)
+
+		// Check that resBytes is not null in order to mitigate appending an empty
+		// object as a result of redundant trailing seperator(s) "---""
+		if !bytes.Equal(resBytes, []byte("null\n")) {
```

Comment:
Rather than checking whether the marshaled result has this somewhat odd string value, would it be better to check resIntf instead before we marshal it?

Reply:
Good suggestion! I have updated the code to check before the object is marshalled!